src/common/FileSystem.cpp

```diff
@@ -1260,31 +1390,60 @@ static void InternalLoadPak(const PakInfo& pak, Util::optional<uint32_t> expecte
 
 	// Load dependencies, but not if a checksum was specified
 	// Do not look for dependencies if it's a legacy pak (pk3)
-	if (!isLegacy && hasDeps && !expectedChecksum) {
-		std::string depsData;
-		if (pak.type == pakType_t::PAK_DIR) {
-			File depsFile = RawPath::OpenRead(Path::Build(pak.path, PAK_DEPS_FILE), err);
-			if (err)
-				return;
-			depsData = depsFile.ReadAll(err);
-			if (err)
-				return;
-		} else if (pak.type == pakType_t::PAK_ZIP) {
-			zipFile.OpenFile(depsOffset, err);
-			if (err)
-				return;
-			offset_t length = zipFile.FileLength(err);
-			if (err)
-				return;
-			depsData.resize(length);
-			auto read = zipFile.ReadFile(&depsData[0], length, err);
-			depsData.resize(read);
-			if (err)
-				return;
-		} else {
-			ASSERT_UNREACHABLE();
+	if (!isLegacy && !expectedChecksum) {
+		if (hasDeleted) {
```

Comment:
I think you don't want this to be guarded by `!expectedChecksum`. I presume the `!expectedChecksum` thing is for the case where the full list of loaded paks is transmitted from the server. You would still need to process the deleted items in that case.

Reply:
fixed (please check this is what expected!).

Reply:
The logic is OK. These 2 if statements could be a single if statement